odc/geo/crs.py

```diff
@@ -309,6 +311,38 @@ def result(x, y):
     def __dask_tokenize__(self):
         return ("odc.geo.crs.CRS", str(self))
 
+    @staticmethod
+    def utm(
```

Comment:
I believe this method handles scenarios when the UTM zone is irregular:

https://pyproj4.github.io/pyproj/stable/examples.html#find-utm-crs-by-latitude-and-longitude

It is used by geopandas and rioxarray in their `estimate_utm_crs` methods.

Reply:
See: https://github.com/geopandas/geopandas/blob/1821b6d72791ce1f7da32db5b22e8803a3dc44b9/geopandas/array.py#L770

Reply:
great, thanks for that, I'll change the logic to use pyproj query